File: frontend/types/enrollment.ts
import { CourseType } from './course';

export interface Course {
  course_code: string;
  course_name: string;
  credits: number;
  year_offered: number;
  course_type: string;
  description: string;
  categories: any | null;
  prerequisites: any | null;
}

export interface Semester {
  semester_id: number;
  year: number;
  semester_number: number;
}

export interface Enrollment {
  enrollment_id: number;
  user_id: number;
  semester_id: number;
  created_at: string;
  semester: Semester;
  courses: Course[];
}

//

export interface EnrollmentType {
  enrollment_id: number;
  user_id: number;
  semester_id: number;
  created_at: string;
  semester: { semester_id: number; year: number; semester_number: number };
  courses: CourseType[];
}

File: backend/internal/dto/enrollment.go
package dto

// CourseInfo holds only the fields we need in the response.
type CourseInfo struct {
    CourseCode string `json:"course_code"`
    CourseName string `json:"course_name"`
    Credits    int    `json:"credits"`
}

// SemesterInfo is embedded in the enrollment DTO.
type SemesterInfo struct {
    Year   int `json:"year"`
    Number int `json:"number"`
}

// EnrollmentHistoryDTO is the API response for each enrollment.
type EnrollmentHistoryDTO struct {
    EnrollmentID uint          `json:"enrollment_id"`
    UserID       uint          `json:"user_id"`
    Semester     SemesterInfo  `json:"semester"`
    Courses      []CourseInfo  `json:"courses"`
}
